criu/cr-restore.c

```diff
@@ -1586,27 +1588,39 @@ static void restore_pgid(void)
 		futex_set_and_wake(&rsti(current)->pgrp_set, 1);
 }
 
+static int __old_mount_proc()
+{
+	char proc_mountpoint[] = "/tmp/crtools-proc.XXXXXX";
+	int fd;
+
+	if (mkdtemp(proc_mountpoint) == NULL) {
+		pr_perror("mkdtemp failed %s", proc_mountpoint);
+		return -1;
+	}
+
+	pr_info("Mount procfs in %s\n", proc_mountpoint);
+	if (mount("proc", proc_mountpoint, "proc", MS_MGC_VAL | MS_NOSUID | MS_NOEXEC | MS_NODEV, NULL)) {
+		pr_perror("mount failed");
+		if (rmdir(proc_mountpoint))
+			pr_err("Unable to remove %s", proc_mountpoint);
```

Comment:
not related to this patch, but while we're at it, this should be `pr_perror()` as `rmdir(2)` sets `errno`.